### PR TITLE
GEOPY-1221: Release version 0.1.0 of octree creation repo

### PR DESCRIPTION
**Pull request by jamiefb**
GEOPY-1221

### CODE REVIEW

**Review by andrewg-mira — DISMISSED** [review]

**Comment by sebhmg** [comment]
@andrewg-mira , can you look at the configuration of badges before publishing?
image

Also `geoapps-utils` in `pyproject.toml` would need to reference the version on PyPI, before that one can go on PyPI

**Review by sebhmg — CHANGES_REQUESTED** [review]
to release and publish on PyPI, need to refence the version of geoapps-utils on PyPI

Also:
- address TODO in readme for license
- add THIRD_PARTY_SOFTWARE.rst

**Comment by domfournier** [comment]
Docs to be addressed on GEOPY-1270

File: .codespellignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 ba
 nd
-astroid
```
Comment (sebhmg) [inline .codespellignore:3]: not necessary anymore: tuned pre-commit to ignore text the lock files

File: README.rst [hunk 2]
```diff
@@ -1,11 +1,11 @@
-|coverage| |maintainability| |precommit_ci| |docs| |style| |version| |status| |pyversions|
```
Comment (sebhmg) [inline README.rst:1]: removing doc for now

File: pylintrc [hunk 3]
```diff
@@ -439,7 +439,9 @@ int-import-graph=
 known-standard-library=
 
 # Force import order to recognize a module as part of a third party library.
-known-third-party=enchant
+known-third-party=enchant,
+                  geoh5py,
+                  geoapps_utils
```
Comment (sebhmg) [inline pylintrc:444]: for some reason necessary for pylint to recognize these as third parties
and stop complaining about the import order

File: pyproject.toml [hunk 4]
```diff
@@ -42,10 +42,20 @@ geoh5py = {version = "~0.8.0rc3", source = "pypi", allow-prereleases = true}
 geoapps-utils = { version = "~0.2.0a2", source = "pypi", allow-prereleases = true}
 
 ## indirect dependencies, forcing them here for installation through Conda not pip
-h5py = "*" # from geoh5py
-Pillow = "*" # from geoh5py
+h5py = "^3.2.1"  # from geoh5py
+Pillow = "^10.0.1"  # from geoh5py
```
Comment (sebhmg) [inline pyproject.toml:46]: if not forcing the version, they sometimes diverge, ending up with one installed by conda and another one with pip through geoph5py

File: pyproject.toml [hunk 5]
```diff
@@ -42,10 +42,20 @@ geoh5py = {version = "~0.8.0rc3", source = "pypi", allow-prereleases = true}
...
+# to be specified to work with conda-lock
+# - from PyPI: my_package = { version = "1.2.3", source = "pypi" }
+# - from URL:
+#   - for a tags:   my_package = { url = "https://github.com/ORGANISATION/REPO/archive/refs/tags/VERSION_TAG.zip#sha256=" }
+#   - for a branch: my_package = { url = "https://github.com/ORGANISATION/REPO/archive/refs/heads/BRANCH.zip#sha256=" }
+# Note - conda-lock does not support the syntax: my_package = { git = ... }
+#        while poetry2conda supports only { git = "...", tag = "..." }, and not { url = ... }
 
 [tool.poetry.dev-dependencies]
+Pygments = "*"
```
Comment (sebhmg) [inline pyproject.toml:58]: to render rst code block in PyCharm

File: pyproject.toml [hunk 6]
```diff
@@ -57,6 +67,7 @@ channels = ['conda-forge']
 
 [tool.isort]
 # settings for compatibility between ``isort`` and ``black`` formatting
+known_third_party = ["geoh5py", "geoapps_utils"]
```
Comment (sebhmg) [inline pyproject.toml:70]: probably not necessary: pylint had an issue but not isort. I will revert that line

File: tests/assets_path_test.py [hunk 7]
```diff
@@ -1,21 +1,17 @@
...
+#  octree-creation-app is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
 
-import octree_creation_app
+from octree_creation_app import assets_path
 
 
 def test_assets_directory_exist():
-    assert assets_path(octree_creation_app.ASSETS_DIR).is_dir()
+    assert assets_path().is_dir()
```
Comment (sebhmg) [inline tests/assets_path_test.py:12]: from the application, we need an easy way to get the assets folder.
Writing is `assets_path(octree_creation_app.ASSETS_DIR)` everytime is somewhat heavy

File: tests/assets_path_test.py [hunk 8]
```diff
@@ -1,21 +1,17 @@
...
-from geoapps_utils.importing import assets_path
+#  octree-creation-app is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
 
-import octree_creation_app
+from octree_creation_app import assets_path
 
 
 def test_assets_directory_exist():
-    assert assets_path(octree_creation_app.ASSETS_DIR).is_dir()
```
Comment (sebhmg) [inline tests/assets_path_test.py:14]: from the application, we need an easy way to get the assets folder.
Writing is `assets_path(octree_creation_app.ASSETS_DIR)` everytime is somewhat heavy

File: octree_creation_app/__init__.py [hunk 9]
```diff
@@ -1,10 +1,20 @@
...
+#  (see LICENSE file at the root of this source code package).
 
 from __future__ import annotations
 
+from pathlib import Path
+
+from geoapps_utils.importing import assets_path as assets_path_impl
+
 __version__ = "0.1.0-rc.1"
-ASSETS_DIR = __file__
```
Comment (sebhmg) [inline octree_creation_app/__init__.py:10]: ASSET_DIR is pointing to a file path: somewhat confusing
I am replacing by a function that uses the base implementation

**Review by domfournier — APPROVED** [review]

**Review by sebhmg — APPROVED** [review]

**Pull request merged**